improver/ensemble_copula_coupling/utilities.py

```diff
@@ -158,31 +158,27 @@ def create_cube_with_percentiles(percentiles, template_cube, cube_data, cube_uni
             template cube.
 
     """
-    percentile_coord = iris.coords.DimCoord(
-        np.float32(percentiles), long_name="percentile", units=unit.Unit("%")
-    )
+    # create cube with new percentile dimension
+    cubes = iris.cube.CubeList([])
+    for point in percentiles:
+        cube = template_cube.copy()
+        cube.add_aux_coord(
+            iris.coords.AuxCoord(
+                np.float32(point), long_name="percentile", units=unit.Unit("%")
+            )
+        )
+        cubes.append(cube)
+    result = cubes.merge_cube()
 
-    metadata_dict = copy.deepcopy(template_cube.metadata._asdict())
-    result = iris.cube.Cube(cube_data, **metadata_dict)
+    # if percentile is single-valued, promote new coordinate to dimension
+    if len(percentiles) == 1:
+        result = iris.util.new_axis(result, result.coord("percentile"))
```

Comment:
Please could you explain why percentiles are initially being created as AuxCoords and then being promoted if single-valued?

Reply:
Good question.  I don't know why the single-value percentile coordinate is required to be dimensional, but that's what this function is providing, and unit tests for the calling class fail if the coordinate is not promoted.  It looked to me like some significant changes to the code might result from addressing this, which I didn't want to add here to an already large PR.  So I'm planning to address this and other issues that might require re-factoring in a follow-on PR.

I'll email you the link to the JIRA ticket on this, I'm planning to collate any outstanding review comments there to address in a follow-on PR.